Usage of deprecated task will break CI after datasets-3.0

Usage of deprecated `task` in `load_dataset`will break CI after datasets-3.0: https://app.circleci.com/pipelines/github/huggingface/transformers/103944/workflows/09136e5e-f232-4132-ba3b-c7b0d5723493/jobs/1382170?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary
```
FAILED examples/tensorflow/test_tensorflow_examples.py::ExamplesTests::test_run_image_classification - TypeError: BuilderConfig.__init__() got an unexpected keyword argument 'task'
```

--- a/examples/tensorflow/image-classification/run_image_classification.py
+++ b/examples/tensorflow/image-classification/run_image_classification.py
@@ -282,7 +282,6 @@ def main():
             data_args.dataset_name,
             data_args.dataset_config_name,
             cache_dir=model_args.cache_dir,
-            task="image-classification",
             token=model_args.token,
             trust_remote_code=model_args.trust_remote_code,
         )
@@ -296,7 +295,6 @@ def main():
             "imagefolder",
             data_files=data_files,
             cache_dir=model_args.cache_dir,
-            task="image-classification",
         )
     # See more about loading any type of standard or custom dataset (from files, python dict, pandas DataFrame, etc) at
     # https://huggingface.co/docs/datasets/loading_datasets.

